Fix git issue in workflow

**Description**

To run any `git` command you should add current path to safe.directory:

```shell
git config --global --add safe.directory ${GITHUB_WORKSPACE}
```

Otherwise you'll receive error:

```text
fatal: detected dubious ownership in repository at '/__w/UTBotJava/UTBotJava'
To add an exception for this directory, call:

	git config --global --add safe.directory /__w/UTBotJava/UTBotJava
Error: Process completed with exit code 128.
```

**To Reproduce**

Steps to reproduce the behavior:

- Run any `git` command in current directory

**Expected behavior**

Git works as expected

**Actual behavior**

`git` fails with exit code 128

**Visual proofs (screenshots, logs, images)**

See [last build](https://github.com/UnitTestBot/UTBotJava/runs/8305239741?check_suite_focus=true)

**Environment**

Not applicable

**Additional context**

Not applicable

--- a/.github/workflows/build-and-run-tests-from-branch.yml
+++ b/.github/workflows/build-and-run-tests-from-branch.yml
@@ -37,6 +37,7 @@ jobs:
       - name: Check out ${{ github.event.inputs.commit_sha }} commit
         if: github.event.inputs.commit_sha != ''
         run: |
+          git config --global --add safe.directory ${GITHUB_WORKSPACE}
           git fetch
           git checkout ${{ github.event.inputs.commit_sha }}
       - id: set-matrix
@@ -69,6 +70,7 @@ jobs:
       - name: Check out ${{ github.event.inputs.commit_sha }} commit
         if: github.event.inputs.commit_sha != ''
         run: |
+          git config --global --add safe.directory ${GITHUB_WORKSPACE}
           git fetch
           git checkout ${{ github.event.inputs.commit_sha }}
       - name: Run monitoring
@@ -123,6 +125,7 @@ jobs:
       - name: Check out ${{ github.event.inputs.commit_sha }} commit
         if: github.event.inputs.commit_sha != ''
         run: |
+          git config --global --add safe.directory ${GITHUB_WORKSPACE}
           git fetch
           git checkout ${{ github.event.inputs.commit_sha }}
       - uses: actions/checkout@v3

